src/components/creator/CreatorWizard.tsx

```diff
@@ -0,0 +1,309 @@
+  }, [taskContents]);
+
+  // Allow use as JSX component
+  return undefined;
+};
+
+const CreatorWizardContext = React.createContext<{
+  errors: CreatorErrors;
+  files: CreatorFiles;
+  onChangeCurrentTask: (index: number | null) => void;
+}>({
+  errors: {
+    taskErrors: new Map(),
+  },
+  files: [],
+  onChangeCurrentTask: () => {},
+});
+
+const TaskErrorPreview = ({ index }: { index: number }) => {
+  const context = useContext(CreatorWizardContext);
+  const error = context.errors.taskErrors.get(index);
+
+  return error !== undefined ? (
+    <pre style={{ whiteSpace: 'pre-wrap' }}>{error}</pre>
+  ) : undefined;
+};
+
+const FileDownload = () => {
+  const { files } = useContext(CreatorWizardContext);
+
+  return (
+    <div>
+      Download these files.
+      {files.map((file) => (
+        <div key={file.name}>
+          <Button
+            variant="secondary"
+            icon={<DownloadIcon />}
+            onClick={() => {
+              const dotIndex = file.name.lastIndexOf('.');
+              const baseName =
+                dotIndex !== -1 ? file.name.substring(0, dotIndex) : file.name;
+              const extension =
+                dotIndex !== -1 ? file.name.substring(dotIndex + 1) : 'txt';
+
+              downloadFile(file.content, baseName, extension);
+            }}
+          >
+            {file.name}
+          </Button>
+
+          <ClipboardCopy
+            isCode
+            isReadOnly
+            variant={ClipboardCopyVariant.expansion}
+            hoverTip="Copy"
+            clickTip="Copied"
+          >
+            {file.content}
+          </ClipboardCopy>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+// Watches for changes in the current step, then calls onChangeCurrentTask so
+// that Creator can update the live preview.
+const WizardSpy = () => {
+  const wizardContext = useContext(DdfWizardContext);
+  const creatorContext = useContext(CreatorWizardContext);
+
+  useEffect(() => {
+    creatorContext.onChangeCurrentTask(
+      taskFromStepName(wizardContext.currentStep.name)
+    );
+  }, [wizardContext.currentStep.name]);
+
+  return undefined;
+};
+
+const CreatorWizard = ({
+  onChangeType,
+  onChangeQuickStartSpec,
+  onChangeBundles,
+  onChangeTaskContents,
+  onChangeCurrentTask,
+  files,
+  errors,
+}: CreatorWizardProps) => {
+  const chrome = useChrome();
+  const schema = useMemo(() => makeSchema(chrome), []);
+
+  const context = useMemo(
+    () => ({
+      errors,
+      files,
+      onChangeCurrentTask,
+    }),
+    [errors, files]
+  );
+
+  const componentMapper = {
+    ...pf4ComponentMapper,
+    'lr-task-error': TaskErrorPreview,
+    'lr-download-files': FileDownload,
+    'lr-wizard-spy': WizardSpy,
+  };
+
+  return (
+    <CreatorWizardContext.Provider value={context}>
+      <FormRenderer
+        onSubmit={() => {}}
+        schema={schema}
+        componentMapper={componentMapper}
+        FormTemplate={FormTemplate}
```

Comment:
In general, you don't want to mix the FormTemplate and child nodes of renderer. In this case, you don't really need the form template. Mostly because you don't need the common form buttons luke submit/cancel. You can get away with just a simple `Form` wrapper and a child function:

```diff
diff --git a/src/components/creator/CreatorWizard.tsx b/src/components/creator/CreatorWizard.tsx
index 5e4d5f2..9bf277a 100644
--- a/src/components/creator/CreatorWizard.tsx
+++ b/src/components/creator/CreatorWizard.tsx
@@ -2,6 +2,7 @@ import {
   Button,
   ClipboardCopy,
   ClipboardCopyVariant,
+  Form,
 } from '@patternfly/react-core';
 import DownloadIcon from '@patternfly/react-icons/dist/dynamic/icons/download-icon';
 import React, { useContext, useEffect, useMemo } from 'react';
@@ -13,7 +14,6 @@ import {
   FormRenderer,
   FormSpy,
 } from '@data-driven-forms/react-form-renderer';
-import { FormTemplate } from '@data-driven-forms/pf4-component-mapper';
 import DdfWizardContext from '@data-driven-forms/react-form-renderer/wizard-context';
 import pf4ComponentMapper from '@data-driven-forms/pf4-component-mapper/component-mapper';
 import {
@@ -276,31 +276,35 @@ const CreatorWizard = ({
         onSubmit={() => {}}
         schema={schema}
         componentMapper={componentMapper}
-        FormTemplate={FormTemplate}
       >
-        <FormSpy subscription={{ values: true }}>
-          {/*
-            In order to display the live preview, we need to update the parent
-            whenever the form state changes. Unfortunately, as best as I can
-            tell, there is no way to pass FormRenderer a callback that's called
-            whenever a value changes.
-
-            The example at [0] shows using a custom component in the schema to
-            watch the values, but it seems clearer to just add it once here
-            (and it avoids introducing another custom component name).
-
-            [0]: https://github.com/data-driven-forms/react-forms/blob/master/packages/react-renderer-demo/src/examples/components/examples/value-listener.js
-             */}
-          {(props) => (
-            <PropUpdater
-              values={props.values}
-              onChangeType={onChangeType}
-              onChangeBundles={onChangeBundles}
-              onChangeQuickStartSpec={onChangeQuickStartSpec}
-              onChangeTaskContents={onChangeTaskContents}
-            />
-          )}
-        </FormSpy>
+        {({ formFields }) => (
+          <Form onSubmit={(e) => e.preventDefault()}>
+            <FormSpy subscription={{ values: true }}>
+              {/*
+                In order to display the live preview, we need to update the parent
+                whenever the form state changes. Unfortunately, as best as I can
+                tell, there is no way to pass FormRenderer a callback that's called
+                whenever a value changes.
+
+                The example at [0] shows using a custom component in the schema to
+                watch the values, but it seems clearer to just add it once here
+                (and it avoids introducing another custom component name).
+
+                [0]: https://github.com/data-driven-forms/react-forms/blob/master/packages/react-renderer-demo/src/examples/components/examples/value-listener.js
+                */}
+              {(props) => (
+                <PropUpdater
+                  values={props.values}
+                  onChangeType={onChangeType}
+                  onChangeBundles={onChangeBundles}
+                  onChangeQuickStartSpec={onChangeQuickStartSpec}
+                  onChangeTaskContents={onChangeTaskContents}
+                />
+              )}
+            </FormSpy>
+            <>{formFields}</>
+          </Form>
+        )}
       </FormRenderer>
     </CreatorWizardContext.Provider>
   );

```